### PR TITLE
Refactors Pirates into Pirate Gangs, Adds the Psyker-gang as new pirates

### PR DESCRIPTION
**Pull request by tralezab**
### Refactor
Pirate gangs are now datumized for extendability, custom dialogue, etc.

### Psyker Gang 🧠 
Psyker-gang Members are pirates who are... yes, Psykers. They're on a gore-binge and need some money for more hits of gore!

- Gore autoinjectors, filled with dirty kronkaine. Don't overdose, you'll go splat.
- Psykerboost armor, reactive armor that refreshes psychic abilities. Given to the leader.

- [x] @Fikou is making the map :D

## Why It's Good For The Game

God I fucking love variety also now we can add as many different pirates as we so desire

<details>
  <summary>Spoiler warning</summary>
  
  image
  
</details>

## Changelog
:cl: Tralezab code, Fikou's map, PigeonVerde and Halcyon for sprites!
add: Psyker-gangers are new pirates
refactor: refactored pirate code so we can add more in the future
/:cl:

### CODE REVIEW

File: code/modules/antagonists/pirate/pirate_gangs.dm [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+	threat_title = "Transfer of goods"
+	threat_content = "Ahoy! This be the %SHIPNAME. Cough up %PAYOFF credits or you'll walk the plank."
+	possible_answers = list("We'll pay.","We will not be extorted.")
+
+	response_received = "Thanks for the credits, landlubbers."
+	response_too_late = "Too late to beg for mercy!"
+	response_not_enough = "Trying to cheat us? You'll regret this!"
+
+///undead skeleton crew looking for booty
```
Comment (Fikou) [inline code/modules/antagonists/pirate/pirate_gangs.dm:106]: fix this

File: code/modules/antagonists/pirate/pirate_gangs.dm [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+/datum/pirate_gang/psykers
+	name = "Roving Psyker-gang"
+
+	ship_template = /datum/map_template/shuttle/pirate/dutchman
+	ship_name_pool = "psyker_names" //just points to THE ONE AND ONLY
```
Comment (Fikou) [inline code/modules/antagonists/pirate/pirate_gangs.dm:111]: you dont have any psyker names

**Review by Fikou — CHANGES_REQUESTED** [review]
where do you give them psyker stuff

File: code/modules/antagonists/pirate/pirate_gangs.dm [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+	response_received = "Thanks for the credits, landlubbers."
+	response_too_late = "Too late to beg for mercy!"
+	response_not_enough = "Trying to cheat us? You'll regret this!"
+
+///undead skeleton crew looking for booty
+/datum/pirate_gang/psykers
+	name = "Roving Psyker-gang"
+
+	ship_template = /datum/map_template/shuttle/pirate/dutchman
```
Comment (Fikou) [inline code/modules/antagonists/pirate/pirate_gangs.dm:110]: set this to psyker

Comment (tralezab) [reply code/modules/antagonists/pirate/pirate_gangs.dm:110]: 👍

**Comment by private-tristan** [comment]
will they still be using echolocation to see? if so they're going to get killed by someone laying down on the ground with a stunbaton (looks about the same as a corpse and they can't use huds/examine to check) because they can't see past 4 tiles (although they are apparently able to show vision?)

**Comment by tralezab** [comment]
> will they still be using echolocation to see? if so they're going to get killed by someone laying down on the ground with a stunbaton (looks about the same as a corpse and they can't use huds/examine to check) because they can't see past 4 tiles (although they are apparently able to show vision?)

they'll make do

**Comment by Overuseddart** [comment]
How do the psykers view the console to drive their ship? They’re blind

**Comment by Fikou** [comment]
image

**Comment by Fikou** [comment]
image

**Comment by calliophis-maculiceps** [comment]
> How do the psykers view the console to drive their ship? They’re blind

image

File: code/game/turfs/closed/wall/misc_walls.dm [hunk 4]
```diff
@@ -125,3 +125,17 @@
...
+	icon_state = "0"
+	plane = PLANE_SPACE
+	turf_flags = NO_RUST
+	smoothing_flags = NONE
+	canSmoothWith = null
+	smoothing_groups = null
+
+/turf/open/wall/space/Initialize(mapload)
+	. = ..()
+	icon_state = SPACE_ICON_STATE(x, y, z)
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/game/turfs/closed/wall/misc_walls.dm:141]: oops

File: code/modules/antagonists/pirate/pirate_event.dm [hunk 5]
```diff
@@ -7,77 +7,67 @@
...
-/proc/spawn_pirates(datum/comm_message/threat, ship_template, skip_answer_check)
+/proc/spawn_pirates(datum/comm_message/threat, datum/pirate_gang/chosen_gang, skip_answer_check)
 	if(!skip_answer_check && threat?.answered == 1)
 		return
 
 	var/list/candidates = poll_ghost_candidates("Do you wish to be considered for pirate crew?", ROLE_TRAITOR)
 	shuffle_inplace(candidates)
 
-	var/datum/map_template/shuttle/pirate/ship = new ship_template
+	var/datum/map_template/shuttle/pirate/ship = new chosen_gang.ship_template
```
[the remark this answers is not in the document]
Comment (tralezab) [reply code/modules/antagonists/pirate/pirate_event.dm]: done! what a pita

**Comment by tralezab** [comment]
Done!!!!!

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 6]
```diff
@@ -786,3 +786,19 @@
...
+	color = "#ffbebe" // kronkaine but with some red
+	ph = 4
+	chemical_flags = NONE
+
+/datum/reagent/drug/kronkaine/gore/overdose_start(mob/living/gored)
+	gored.visible_message(
+		span_danger("[gored] explodes in a shower of gore!"),
+		span_userdanger("GORE! GORE! GORE! YOU'RE GORE! TOO MUCH GORE! YOU'RE GORE! GORE! IT'S OVER! GORE! GORE! YOU'RE GORE! TOO MUCH G-"),
+	)
+	new /obj/structure/bouncy_castle(gored.loc, gored)
```
Comment (Kylerace) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:803]: guh

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 7]
```diff
@@ -97,32 +97,38 @@
...
-	icon_state = "detective"
+/obj/item/gun/ballistic/revolver/c38
+	name = "\improper .38 revolver"
+	desc = "A classic, if not outdated, lethal firearm. Uses .38 Special rounds."
 	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/rev38
 	initial_caliber = CALIBER_38
-	alternative_caliber = CALIBER_357
+	icon_state = "c38"
+	fire_sound = 'sound/weapons/gun/revolver/shot.ogg'
 	initial_fire_sound = 'sound/weapons/gun/revolver/shot.ogg'
```
Comment (Fikou) [inline code/modules/projectiles/guns/ballistic/revolver.dm:107]: these initial variables should be on the detective subtype. they only matter for weapons with multiple modes

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 6 above)
Comment (Fikou) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm:803]: image

File: code/modules/religion/honorbound/honorbound_trauma.dm [hunk 8]
```diff
@@ -183,7 +183,7 @@
  */
 /datum/brain_trauma/special/honorbound/proc/punishment(mob/living/carbon/human/user, school)
 	switch(school)
-		if(SCHOOL_HOLY, SCHOOL_MIME, SCHOOL_RESTORATION)
+		if(SCHOOL_HOLY, SCHOOL_MIME, SCHOOL_RESTORATION, SCHOOL_PSYCHIC, SCHOOL_UNSET)
```
Comment (Fikou) [inline code/modules/religion/honorbound/honorbound_trauma.dm:186]:
```suggestion
		if(SCHOOL_UNSET, SCHOOL_HOLY, SCHOOL_MIME, SCHOOL_RESTORATION, SCHOOL_PSYCHIC)
```
defaults should be at the beginning

**Review by Kylerace — APPROVED** [review]

File: code/datums/components/echolocation.dm [hunk 9]
```diff
@@ -108,6 +113,7 @@
 	final_image.loc = images_are_static ? get_turf(input) : input
 	final_image.dir = input.dir
 	final_image.alpha = 0
+	final_image.appearance_flags |= PASS_MOUSE
```
Comment (Fikou) [inline code/datums/components/echolocation.dm:116]:
```suggestion
	final_image.alpha = 0
```

**Comment by TheSmallBlue** [comment]
Played as a psyker pirate a bit ago, holy shit it sucked ass. The point of being a psyker chaplain is that you have a rite that straight up summons weapons, psyker pirates dont have anything like that at all, in fact their ship starts with a stupidly limited amount of weapons, and a lot of ammo, of which they can't carry any because their pockets is already full with gore injectors.
It's just awful in every sense of the word.

**Pull request merged**